src/Worker.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Equip\Queue;
+
+use Equip\Queue\Driver\DriverInterface;
+use Exception;
+
+class Worker
+{
+    /**
+     * @var DriverInterface
+     */
+    private $driver;
+
+    /**
+     * @var Event
+     */
+    private $event;
+
+    /**
+     * @var array
+     */
+    private $handlers;
+
+    /**
+     * @var array
+     */
+    private $options = [];
+
+    /**
+     * @param DriverInterface $driver
+     * @param Event $event
+     * @param array $handlers
+     * @param array $options
+     */
+    public function __construct(
+        DriverInterface $driver,
+        Event $event,
+        array $handlers = [],
+        array $options = []
+    ) {
+        $this->driver = $driver;
+        $this->event = $event;
+        $this->handlers = $handlers;
+        $this->options = array_merge($this->options, $options);
+    }
+
+    /**
+     * Consumes messages off of the queue
+     *
+     * @param string $queue
+     */
+    public function consume($queue)
+    {
+        declare (ticks = 1);
+
+        while ($this->tick($queue)) { /* NOOP */ }
+    }
+
+    /**
+     * Handles fetching messages from the queue
+     *
+     * @param string $queue
+     *
+     * @return bool
+     */
+    private function tick($queue)
+    {
+        $packet = $this->driver->pop($queue);
+
+        $message = json_decode($packet, true);
+        if (empty($message)) {
+            return true;
+        }
+
+        $handler = $this->getHandler($message['name'], $this->handlers);
+        if (!$handler) {
+            return true;
+        }
+
+        if (!is_callable($handler)) {
+            return true;
+        }
+
+        try {
+            call_user_func($handler, $message);
```

Comment:
Do we want to allow handles to terminate the worker process by returning their return value, so that they can return `false` to terminate the loop in `consume()`?